Sources/SocksCore/Conversions.swift

```diff
@@ -20,9 +20,30 @@ extension Array {
 }
 
 extension String {
-    
     public func toBytes() -> [UInt8] {
         return Array(self.utf8)
     }
 }
 
+// Sendable and Receivable implementations
+
+extension String: Sendable {
+    public func serialize() -> [UInt8] {
+        return self.toBytes()
+    }
+}
+
+extension String: Receivable {
+    static func deserialize(reader: (maxBytes: Int) throws -> [UInt8]) throws -> String {
```

Comment:
@czechboy0 I personally prefer serialize/deserialize, but I've lost the argument most of the time, so now I seek out continuity. Whatever ends up here is fine by me, not a big deal either way.

For `@noescape`, part of my reason for suggesting was that the function actually has a return. It seems counterintuitive to have a closure doing the reading asynchronously beyond the scope of the return. 

Am I missing something on the secondary part?